src/Sarif.Multitool/TransformCommand.cs

```diff
@@ -58,5 +81,28 @@ public static int Run(TransformOptions transformOptions)
 
             return 0;
         }
+
+        private string SniffVersion(string sarifPath, int tokenCountLimit = int.MaxValue)
+        {
+            int tokenCount = 0;
+
+            // When we're unit-testing, we'll retrieve a string representation of the file contents and pass this to a stream reader instance.
+            TextReader textReader = _testing ? new StreamReader(new MemoryStream(Encoding.UTF8.GetBytes(_fileSystem.ReadAllText(sarifPath)))) : new StreamReader(sarifPath);
+
+            using (JsonTextReader reader = new JsonTextReader(textReader))
+            {
+                while (reader.Read() && tokenCount < tokenCountLimit)
```

Comment:
>tokenCountLimit [](start = 53, length = 15)

I understand the sentiment, but I don't think we should limit it. Some implementations ignore the spec's advice to put `version` first. Consider the file that Aaron just sent us. It emits the properties in alphabetical order, so `version` comes last. If such an implementation produces a large file, we'll never get to `version`.

Reply:
In Aaron's case, his file has fewer than  2,147,483,647 JSON tokens and so we do end up finding his version. :) this configurability isn't exposed anywhere currently, so i could just disable it entirely for now.

---
In reply to: [229512144](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229512144) [](ancestors = 229512144)

Reply:
Yes, I think we should. My point wasn't that Aaron's file was too large; it was that since he emitted `version` last, his implementation was _in danger of_ creating a file that you can't transform.

---
In reply to: [229513592](https://github.com/Microsoft/sarif-sdk/pull/1125#discussion_r229513592) [](ancestors = 229513592,229512144)